extensions/sql-database-projects/src/test/project.test.ts

```diff
@@ -1253,6 +1253,96 @@ describe('Project: sdk style project content operations', function (): void {
 		should(project.files.find(f => f.relativePath === 'folder3\\')!).not.equal(undefined, 'folder3\\ should be loaded');
 		should(project.files.find(f => f.relativePath === 'folder3\\innerFolder\\')!).not.equal(undefined, 'folder3\\innerFolder\\ should be loaded');
 	});
+
+	it('Should handle deleting empty folders', async function (): Promise<void> {
+		const testFolderPath = await testUtils.generateTestFolderPath();
+		projFilePath = await testUtils.createTestSqlProjFile(baselines.newSdkStyleProjectSdkNodeBaseline, testFolderPath);
+
+		const project: Project = await Project.openProject(projFilePath);
+
+		should(project.files.filter(f => f.type === EntryType.File).length).equal(0);
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(0);
+
+		// add an empty folder
+		await project.addFolderItem('folder1');
+
+		// verify folder was added
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(1);
```

Comment:
Not looking for any action on your part, just musing... but it's probably time to track files and folders separately to avoid having to do this sort of filtering.

Reply:
yeah I've been thinking that every time I make a change for the SDK style projects, but wasn't sure if there was a reason why they were grouped together.

Reply:
I think it's a vestige from when I was hacking out reading the old-style sqlproj format to get the initial version of the extension working.